### ISSUE TITLE
Capture of graphics trace fails as app starts up

### ISSUE THREAD

**Issue by adowdy** [google/gapid#2000]
I have a Kindle 7th gen device with debugging correctly configured in dev options, I was able to use the 
 [...] button to find the main activity for Package/Action as such:
android.intent.action.MAIN:arrow.blah.com.arrow/arrow.blah.com.arrow.ui.main.MainActivity
My main activity uses an GLSurfaceView, which I call to native GL to render from GLSurfaceView.Renderer interface implementation only.

What I see is the capture start, claims to begin capturing for a few messages (screen is completely white), then just says EOF and doesn't go further. At that point, the app on the device just closes itself. Has anyone seen this issue?

Log below:

```
Press enter to stop capturing...
00:32:31.760 I: [Try ABI: armeabi-v7a⇒gapidapk.EnsureInstalled] <gapit> Examining gapid.apk on host...
00:32:31.764 I: [Try ABI: armeabi-v7a⇒gapidapk.EnsureInstalled] <gapit> Looking for gapid.apk...
00:32:32.212 I: [Try ABI: armeabi-v7a⇒gapidapk.EnsureInstalled] <gapit> Found gapid package...
00:32:32.212 I: [startDevInfoService] <gapit> Attempt to start service: com.google.android.gapid.DeviceInfoService
00:32:32.860 I: <gapit> Adding new device
00:32:32.860 I: <gapit> Device list:
00:32:32.860 I: <gapit>   G0W0MA077444F035
00:32:33.736 I: <gapit> Package is debuggable
00:32:33.782 I: [start] <gapit> Turning device screen on
00:32:33.806 I: [start] <gapit> Checking for lockscreen
00:32:33.830 I: [start] <gapit> Checking gapid.apk is installed
00:32:33.830 I: [start] <gapit> Forwarding
00:32:33.832 I: [start] <gapit> Starting activity in debug mode
00:32:34.781 I: [start] <gapit> Forwarding TCP port 39894 -> JDWP pid 11025
00:32:34.783 I: [start] <gapit> Connecting to JDWP
00:32:34.786 I: [start] <gapit> Waiting for ApplicationLoaders.getClassLoader()
00:32:34.788 W: [start] <gapit> Couldn't break in ApplicationLoaders.getClassLoader. Vulkan will not be supported.
00:32:34.788 I: [start] <gapit> Waiting for Application Creation
00:32:34.789 I: [start] <gapit>    Waiting for Application.<init>()
00:32:38.103 I: [start] <gapit>    Waiting for arrow.blah.com.arrow.ArrowApplication.onCreate()
00:32:38.477 I: [start] <gapit> Installing interceptor libraries
00:32:39.100 I: [start] <gapit> GVR library not found
00:32:39.169 I: [start] <gapit> Waiting for connection to localhost:37312...
00:32:41.110 W: [start] <gapit> Failed to read packet. Error: read tcp 127.0.0.1:38444->127.0.0.1:39894: use of closed network connection
00:32:41.112 I: <gapit> Creating file '/home/adowdy/arrow_20180618_1731.gfxtrace'
00:32:41.612 I: <gapit> Capturing: 16B in 0s
00:32:42.612 I: <gapit> Capturing: 16B in 1s
00:32:43.613 I: <gapit> Capturing: 16B in 2s
00:32:44.613 I: <gapit> Capturing: 16B in 3s
00:32:44.978 I: <gapit> EOF: 16B

```

EDIT ----------

I'm digging through logcat to try and find something interesting... this is what looks suspicious

./adb logcat | grep -i gapid
```
I/GAPID   (12547): [gapii/cc/spy.cpp:58] JNI_OnLoad() was called. vm = 0xb7865920
I/GAPID   (12547): [gapii/cc/spy.cpp:122] Constructing spy...
I/GAPID   (12547): [gapii/cc/connection_stream.cpp:35] GAPII awaiting connection on pipe gapii (abstract)
I/GAPID   (12547): [gapii/cc/spy.cpp:158] Connection made
I/GAPID   (12547): [gapii/cc/spy.cpp:165] Connection header read
E/GAPID   (12547): [gapii/cc/spy.cpp:180] APIS 0000000a
I/GAPID   (12547): [gapii/cc/spy.cpp:181] GAPII connection established. Settings:
I/GAPID   (12547): [gapii/cc/spy.cpp:182] Observe framebuffer every 0 frames
I/GAPID   (12547): [gapii/cc/spy.cpp:183] Observe framebuffer every 0 draws
I/GAPID   (12547): [gapii/cc/spy.cpp:184] Disable precompiled shaders: false
I/GAPID   (12547): [gapii/cc/android/installer.cpp:128] Installing GAPII hooks...
I/GAPID   (12547): [gapii/cc/android/installer.cpp:150] Interceptor functions resolved
I/GAPID   (12547): [gapii/cc/android/installer.cpp:152] Calling gInitializeInterceptor at 0xa2225c83...
I/GAPID   (12547): [gapii/cc/android/installer.cpp:157] Installing OpenGL ES hooks...
W/GAPID   (12547): [gapii/cc/android/installer.cpp:120] Interceptor error: Intercepting function at 0xb6d3af64 failed: End of function reached after 4 byte when rewriting 8 bytes
E/GAPID   (12547): [gapii/cc/android/installer.cpp:241] Couldn't intercept function glVertexPointerBounds at 0xb6d3af64
W/GAPID   (12547): [gapii/cc/android/installer.cpp:120] Interceptor error: Intercepting function at 0xb6d3b5b0 failed: End of function reached after 4 byte when rewriting 8 bytes
E/GAPID   (12547): [gapii/cc/android/installer.cpp:241] Couldn't intercept function glPointSizePointerOESBounds at 0xb6d3b5b0
W/GAPID   (12547): [gapii/cc/android/installer.cpp:120] Interceptor error: Intercepting function at 0xb6d318dc failed: Unhandled instruction: 	nop (OpcodeId: 220)
E/GAPID   (12547): [gapii/cc/android/installer.cpp:241] Couldn't intercept function glReadPixels at 0xb6d318dc
W/GAPID   (12547): [gapii/cc/android/installer.cpp:120] Interceptor error: Intercepting function at 0xb6d3acf0 failed: End of function reached after 4 byte when rewriting 8 bytes
E/GAPID   (12547): [gapii/cc/android/installer.cpp:241] Couldn't intercept function glNormalPointerBounds at 0xb6d3acf0
```

EDIT 2 ---

Just confirmed I can successfully capture traces on a different device (Nexus 5X) with the exact same system and host configuration as the other non-working one (Kindle Fire gen 7). Perhaps not all devices are supported?

**Comment by AWoloszyn** [comment]
Thanks for the report. We try to support most devices, but sometimes drivers will do things that are unexpected. 

W/GAPID   (12547): [gapii/cc/android/installer.cpp:120] Interceptor error: Intercepting function at 0xb6d3af64 failed: End of function reached after 4 byte when rewriting 8 bytes

@ben-clayton  This looks like our old friend PLT again.

**Comment by nerael** [comment]
No problem. Really great tool you guys provide here, I was able to use it on other devices :)

**Comment by AWoloszyn** [comment]
A bunch of PLT fixes went in for 1.4.0. This should be fixed now.
